# Step-1: Collection of Data
The data set contains responses from 1453 adult Australian consumers relating to
their perceptions of McDonald’s with respect to the following attributes: YUMMY,
CONVENIENT, SPICY, FATTENING, GREASY, FAST, CHEAP, TASTY, EXPENSIVE, HEALTHY, and DISGUSTING.

Source: https://homepage.boku.ac.at/leisch/MSA/datasets/mcdonalds.csv

In [41]:
#importing the required packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [42]:
#reading data
df = pd.read_csv("mcdonalds.csv")

In [43]:
df.head()

yummy convenient spicy fattening greasy fast cheap tasty expensive healthy  \
0    No        Yes    No       Yes     No  Yes   Yes    No       Yes      No   
1   Yes        Yes    No       Yes    Yes  Yes   Yes   Yes       Yes      No   
2    No        Yes   Yes       Yes    Yes  Yes    No   Yes       Yes     Yes   
3   Yes        Yes    No       Yes    Yes  Yes   Yes   Yes        No      No   
4    No        Yes    No       Yes    Yes  Yes   Yes    No        No     Yes   

  disgusting Like  Age      VisitFrequency  Gender  
0         No   -3   61  Every three months  Female  
1         No   +2   51  Every three months  Female  
2         No   +1   62  Every three months  Female  
3        Yes   +4   69         Once a week  Female  
4         No   +2   49        Once a month    Male

In [44]:
# Converting data of Yes/No data into numerical inputs
# Yes == 1
# No == 0

df["yummy"] = (df["yummy"] == "Yes").astype(int)
df["convenient"] = (df["convenient"] == "Yes").astype(int)
df["spicy"] = (df["spicy"] == "Yes").astype(int)
df["fattening"] = (df["fattening"] == "Yes").astype(int)
df["greasy"] = (df["greasy"] == "Yes").astype(int)
df["fast"] = (df["fast"] == "Yes").astype(int)
df["cheap"] = (df["cheap"] == "Yes").astype(int)
df["tasty"] = (df["tasty"] == "Yes").astype(int)
df["expensive"] = (df["expensive"] == "Yes").astype(int)
df["healthy"] = (df["healthy"] == "Yes").astype(int)
df["disgusting"] = (df["disgusting"] == "Yes").astype(int)

In [45]:
df.head()

yummy  convenient  spicy  fattening  greasy  fast  cheap  tasty  expensive  \
0      0           1      0          1       0     1      1      0          1   
1      1           1      0          1       1     1      1      1          1   
2      0           1      1          1       1     1      0      1          1   
3      1           1      0          1       1     1      1      1          0   
4      0           1      0          1       1     1      1      0          0   

   healthy  disgusting Like  Age      VisitFrequency  Gender  
0        0           0   -3   61  Every three months  Female  
1        0           0   +2   51  Every three months  Female  
2        1           0   +1   62  Every three months  Female  
3        0           1   +4   69         Once a week  Female  
4        1           0   +2   49        Once a month    Male

In [46]:
# Mean of the columns with Yes or No

round(df[["yummy","convenient","spicy","fattening","greasy","fast","cheap","tasty","expensive","healthy","disgusting"]].mean(),2)

yummy         0.55
convenient    0.91
spicy         0.09
fattening     0.87
greasy        0.53
fast          0.90
cheap         0.60
tasty         0.64
expensive     0.36
healthy       0.20
disgusting    0.24
dtype: float64

#### Insights
| Attribute   | % of Yes |
|-------------|----------|
| Yummy       | 55%      |
| Convenient  | 91%      |
| Spicy       | 09%      |
| Fattening   | 87%      |
| Greasy      | 53%      |
| Fast        | 90%      |
| Cheap       | 60%      |
| Tasty       | 64%      |
| Expensive   | 36%      |
| Healthy     | 20%      |
| Disgusting  | 24%      |